pkg/endpoints/pipeline.go

```diff
@@ -59,6 +67,25 @@ type BuildRequest struct {
 	BRANCH   string `json:"branch"`
 }
 
+// ResourceBinding - a name and a reference to a resource
+type ResourceBinding struct {
+	BINDINGNAME     string `json:"bindingname"`
+	RESOURCEREFNAME string `json:"resourcerefname"`
+}
+
+//ManualPipelineRun - represents the data required to create a new PipelineRun
+type ManualPipelineRun struct {
+	PIPELINENAME      string `json:"pipelinename"`
+	PIPELINERUNTYPE   string `json:"pipelineruntype,omitempty"`
+	GITRESOURCENAME   string `json:"gitresourcename,omitempty"`
+	IMAGERESOURCENAME string `json:"imageresourcename,omitempty"`
+	GITCOMMIT         string `json:"gitcommit,omitempty"`
+	REPONAME          string `json:"reponame,omitempty"`
+	REPOURL           string `json:"repourl,omitempty"`
+	HELMSECRET        string `json:"helmsecret,omitempty"`
+	REGISTRYSECRET    string `json:"registrysecret,omitempty"`
```

Comment:
Sorry; this is another question about the scope of the PipelineRun creation 😬

Is there a reason why we don't want to specify the `SERVICEACCOUNT` in the `ManualPipelineRun` struct? This is a pretty strict design decision to say that the user can only create a PipelineRun using one service account. Since we only use the `PIPELINE_RUN_SERVICE_ACCOUNT` here https://github.com/tektoncd/dashboard/pull/33/files#diff-2a268056787157d73c4168498850de86R240, it seems to me like the code would function the same way if the user was able to specify whatever service account they want for the PipelineRun. Then we would also be able to remove the restriction of one `PIPELINE_RUN_SERVICE_ACCOUNT`.

However, please let me know if there is a benefit to restricting the user to one service account that I am not seeing at the moment. I know this used to be a bigger PR, so there might be a bigger-picture that I'm missing out on.

Reply:
I agree! especially as ServiceAccount is something that can be defined in the spec for a PipelineRun, I will remove the env variable and take ServiceAccount as a field for the struct 👍